packages/react-flicking/src/react-flicking/types.ts

```diff
@@ -0,0 +1,27 @@
+import NativeComponent from "@egjs/component";
+import { ChangeEvent, FlickingEvent, SelectEvent, Plugin, NeedPanelEvent, FlickingStatus } from "@egjs/flicking";
+import NativeFlicking from "@egjs/flicking";
+
+export type ExcludeKeys = keyof NativeComponent | "replace" | "append" | "remove" | "prepend" | "sync" | "getCloneCount";
+export type PureFlicking = Pick<NativeFlicking, Exclude<keyof NativeFlicking, ExcludeKeys>>;
+export type FlickingType<T> = {
+  [key in keyof PureFlicking]:
+    PureFlicking[key] extends (...params: any) => NativeFlicking ?
+      (...params: Parameters<PureFlicking[key]>) => T : PureFlicking[key]
+};
```

Comment:
If this can be done, I hope there's way that map NativeFlicking's public functions in `FlickingType` to React Flicking automatically.
Manually managing public methods per frameworks looks bit harsh for me.

Reply:
@WoodNeck 
I think it's going to be difficult right now. FlickeringType is just a type, so I think there will be a type issue if we approach proteotype.